lib/execjs/ruby_rhino_runtime.rb

```diff
@@ -32,7 +32,11 @@ def eval(source, options = {})
       end
 
       def call(properties, *args)
-        unbox @rhino_context.eval(properties).call(*args)
+        # Might no longer be necessary if therubyrhino handles Symbols directly:
+        # https://github.com/rubyjs/therubyrhino/issues/43
+        converted_args = args.map { |arg| Symbol === arg ? arg.to_s : arg }
```

Comment:
This isn't quite enough thoug. `args` could be `[{foo: 42}]`, so you'd need something that walk down the object graph.

An easy fix could be `JSON.parse(JSON.dump(args))`.

Reply:
Oh, of course, not sure how I missed that.
I'll do that then, it should provide better compatibility for Rhino, at the cost of some performance but then anyway it's not a very common JS runtime nowadays (very old JS version).

Reply:
Done, I used `JSON.parse(JSON.generate(args), create_additions: false)` for consistency with ExternalRuntime, and added tests like this nested example.

Reply:
> at the cost of some performance

Yeah, it's gonna be a bit slower, but ExecJS is really not what you want to use if you want the best performance.